broker.go

```diff
@@ -675,69 +716,86 @@ func (b *Broker) muCloseDeadConnection(conn *connection) {
 // offset will return offset value for given partition. Use timems to specify
 // which offset value should be returned.
 func (b *Broker) offset(topic string, partition int32, timems int64) (offset int64, err error) {
-	conn, err := b.muLeaderConnection(topic, partition)
-	if err != nil {
-		return 0, err
-	}
-	resp, err := conn.Offset(&proto.OffsetReq{
-		ClientID:  b.conf.ClientID,
-		ReplicaID: -1, // any client
-		Topics: []proto.OffsetReqTopic{
-			{
-				Name: topic,
-				Partitions: []proto.OffsetReqPartition{
-					{
-						ID:         partition,
-						TimeMs:     timems,
-						MaxOffsets: 2,
+	var resErr error
+	for retry := 0; retry < b.conf.RetryErrLimit; retry++ {
```

Comment:
I was thinking about it and I still don't see the right way to do it.  We can make small improvements and standardize code in Producer, Consumer and Broker ( or even merge everything in a Broker)  but I don't see how we can localize code for retrying in one place.  I would be really happy if you shared your ideas about this.